src/main/java/com/atlassian/plugins/confluence/markdown/MarkdownFromURLMacro.java

```diff
@@ -56,15 +65,104 @@
 public class MarkdownFromURLMacro extends BaseMacro implements Macro {
 
     private final XhtmlContent xhtmlUtils;
+    private BandanaManager bandanaManager;
+    private ConfluenceBandanaContext context = new ConfluenceBandanaContext("markdown-plugin");
+    private ObjectMapper objectMapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    private LinkedList<String[]> whitelist;
+    private boolean enabled;
 
     private PageBuilderService pageBuilderService;
+    
+    class PrivateRepositoryException extends Exception {
+		public PrivateRepositoryException(String message) {
+			super(message);
+		}
+	}
+	
+	class NonWhitelistURLException extends Exception {
+		public NonWhitelistURLException() {
+			super();
+		}
+	}
+	
+	class IllegalRedirectException extends Exception {
+		boolean isMalformed;
+		
+		public IllegalRedirectException() {
+			super();
+		}
+		
+		public IllegalRedirectException(boolean isMalformed, String message) {
+			super(message);
+			this.isMalformed = isMalformed;
+		}
+	}
 
     @Autowired
-    public MarkdownFromURLMacro(@ComponentImport PageBuilderService pageBuilderService, XhtmlContent xhtmlUtils) {
+    public MarkdownFromURLMacro(@ComponentImport PageBuilderService pageBuilderService, XhtmlContent xhtmlUtils, BandanaManager bandanaManager) {
         this.pageBuilderService = pageBuilderService;
         this.xhtmlUtils = xhtmlUtils;
+        this.bandanaManager = bandanaManager;
+    }
+    
+    private void initWhitelistConfiguration() {
+        MacroConfigModel model = new MacroConfigModel();
+        String config = (String) this.bandanaManager.getValue(context, PLUGIN_CONFIG_KEY);
+        if (config != null) {
+            try {
+                model = objectMapper.readValue(config, MacroConfigModel.class);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        enabled = model.getConfig().getEnabled();
+        whitelist = new LinkedList<String[]>();
+        for (String domain : model.getConfig().getWhitelist()) {
+        	if (domain.length() == 0) continue;
+        	String[] domainComponents = domain.split("\\.");
+        	whitelist.add(domainComponents);
+        }
+    }
+    
+    private boolean isAllowedToProceed(URL url) {
```

Comment:
Needs to support IPs in the whitelist. Then we also need to adjust the copy on the frontend to mention that we support domains and IPs.